optuna/multi_objective/_hypervolume/wfg.py

```diff
@@ -20,8 +20,8 @@ def __init__(self) -> None:
         self._reference_point: Optional[np.ndarray] = None
 
     def _compute(self, solution_set: np.ndarray, reference_point: np.ndarray) -> float:
-        self._reference_point = reference_point.copy()
-        return self._compute_rec(solution_set)
+        self._reference_point = reference_point.copy().astype(np.float64)
```

Comment:
`astype` returns a new array by default.

```suggestion
        self._reference_point = reference_point.astype(np.float64)
```

Reply:
Thanks!